### PR TITLE
applayer/htp: convert to new FAIL/PASS API-v6

### PR DESCRIPTION
**Pull request by Nancyenos**
Ticket: #6935

Make sure these boxes are checked accordingly before submitting your Pull Request -- thank you.

## Contribution style:
- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html

## Our Contribution agreements:
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)

Link to ticket: https://redmine.openinfosecfoundation.org/issues/6935

Describe changes:
-convert applayer/htp: convert to new FAIL/PASS API
-previous pr #11973

### CODE REVIEW

File: src/app-layer-htp.c [hunk 1]
```diff
@@ -4264,82 +3983,53 @@ libhtp:\n\
...
-    else {
-        printf("Failed to parse address: %s\n", addr);
-        goto end;
-    }
+    FAIL_IF(inet_pton(AF_INET, addr, buf) != 1);
+    (void)SCRadixFindKeyIPV4BestMatch(buf, cfgtree, &user_data);
+    FAIL_IF_NULL(user_data);
+    HTPCfgRec *htp_cfg_rec = user_data;
+    FAIL_IF_NULL(htp);
+    htp = htp_cfg_rec->cfg;
```
Comment (jufajardini) [inline src/app-layer-htp.c:4001]: Juding from the original test, it seems to make more sense to invert these two lines, that is, have the check for `htp == NULL` _after_ we've assigned the new value to it.
(This one is on me, I see that I suggest it just as you did. Sorry about that)

File: src/app-layer-htp.c [hunk 2]
```diff
@@ -4264,82 +3983,53 @@ libhtp:\n\
...
-        goto end;
-    }
-
-    ret = 1;
+    FAIL_IF(inet_pton(AF_INET6, addr, buf) != 1);
+    (void)SCRadixFindKeyIPV6BestMatch(buf, cfgtree, &user_data);
+    FAIL_IF_NULL(user_data);
+    htp_cfg_rec = user_data;
+    FAIL_IF_NULL(htp);
+    htp = htp_cfg_rec->cfg;
```
Comment (jufajardini) [inline src/app-layer-htp.c:4011]: same here.

**Review by jufajardini — CHANGES_REQUESTED** [review]
We're almost there, good job with such a big first contrib!

I think that if you address the three inline comments that I've left that require changes, the next version should be ready for merging :)

**Comment by Nancyenos** [comment]
> We're almost there, good job with such a big first contrib!
> 
> I think that if you address the three inline comments that I've left that require changes, the next version should be ready for merging :)

Let me address then just now, cant waut to see this contribution merged! its been a long road

**Comment by Nancyenos** [comment]
work continued in #12023

**Pull request closed without merging**